### ISSUE TITLE
ReferenceError: Buffer is not defined

### ISSUE THREAD

**Issue by andrzej-casper** [CasperDash/useWallet#70]
I am trying to use `useWallet` in default React app:

```tsx
import logo from './logo.svg';
import './App.css';

import {
  CasperDashConnector,
  CasperSignerConnector,
  CasperProvider,
  createClient
} from '@casperdash/usewallet';

const client = createClient({
  connectors: [new CasperSignerConnector(), new CasperDashConnector()],
  autoConnect: true,
});

function App() {
  return (
    <CasperProvider client={client}>
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
            >
            Learn React
          </a>
        </header>
      </div>
    </CasperProvider>
  );
}

export default App;
```

I am getting the following error:

```
Buffer is not defined
ReferenceError: Buffer is not defined
    at ./node_modules/@ledgerhq/devices/lib-es/hid-framing.js (http://localhost:3000/static/js/bundle.js:325:9)
    at options.factory (http://localhost:3000/static/js/bundle.js:76988:31)
    at __webpack_require__ (http://localhost:3000/static/js/bundle.js:76381:33)
    at fn (http://localhost:3000/static/js/bundle.js:76645:21)
    at ./node_modules/@ledgerhq/hw-transport-webusb/lib-es/TransportWebUSB.js (http://localhost:3000/static/js/bundle.js:3381:87)
    at options.factory (http://localhost:3000/static/js/bundle.js:76988:31)
    at __webpack_require__ (http://localhost:3000/static/js/bundle.js:76381:33)
    at fn (http://localhost:3000/static/js/bundle.js:76645:21)
    at ./node_modules/@casperdash/usewallet-core/dist/index.js (http://localhost:3000/static/js/bundle.js:70409:87)
    at options.factory (http://localhost:3000/static/js/bundle.js:76988:31)
```

It seems that `@ledgerhq/hw-transport-webusb` package should not be used in the frontend. See https://github.com/LedgerHQ/ledgerjs/issues/734.

---

Even when I fixed it manually, I started getting some BigInt errors. This package is also not compatible with CommonJS, so it is a bit problematic to use. **Does it even work?**

**Comment by hoailinh1210** [comment]
Hi @andrzej-casper,
If you use vitejs for building you can ref my config to load buffer from node-polyfills
```
import path from 'path';

import react from '@vitejs/plugin-react-swc';
import { defineConfig } from 'vite';
import { nodePolyfills } from 'vite-plugin-node-polyfills';

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    react(),
    nodePolyfills({
      // Whether to polyfill `node:` protocol imports.
      protocolImports: true,
    }),
  ],
  resolve: {
    alias: {
      '@': path.resolve(__dirname, './src'),
    },
  },
});
```

>  I started getting some BigInt errors. 

Could you please provide me with more details about this point?

> This package is also not compatible with CommonJS, so it is a bit problematic to use. Does it even work?

I will need to check on it and revert soon

**Comment by andrzej-casper** [comment]
@hoailinh1210 I am using the default CRA bundler - Webpack - however, this library should not require additional setup. I think you could include `Buffer` polyfill when building it, so NPM package will work out of the box.

I was getting some BigInt errors after I got rid of ledger code; I immediately gave up on useWallet and removed everything :sweat:.

**Comment by hoailinh1210** [comment]
Hey @andrzej-casper, 
You're correct that it "should not require additional setup". We'll be polyfilling the `buffer` in the next release. 

Thanks for your feedback.

**Comment by hoailinh1210** [comment]
Hey @andrzej-casper, 
I want to let you know that I have separated the `ledger connector` into a single package and poly-filled buffer. This means that if you use `useWallet` it will work correctly.
With the ledger, you will need to install other packages (@casperdash/usewallet-ledger) and use the ledger connector to inject.
I will close this topic after

**Comment by hoailinh1210** [comment]
Resolved `buffer` on @casperdash/usewallet@1.0.0

**Comment by andrzej-casper** [comment]
@hoailinh1210 Awesome, I just tried new version and integration with plain CRA worked out of the box :tada:
